internal/exec/exec.go

```diff
@@ -62,31 +61,23 @@ type fieldToExec struct {
 	out      *bytes.Buffer
 }
 
-// TODO: Don't want to pass request here, but need the directive visitors exposed on the field
-func (f *fieldToExec) resolve(ctx context.Context, r *Request) (output interface{}, err error) {
+func (f *fieldToExec) resolve(ctx context.Context) (output interface{}, err error) {
 	var args interface{}
 
 	if f.field.ArgsPacker != nil {
 		args = f.field.PackedArgs.Interface()
 	}
 
-	n := len(f.field.Directives)
-
-	if n == 0 {
-		return f.resolveField(ctx, args)
-	}
-
 	currResolver := f.resolveField
 
-	for i := n - 1; i >= 0; i-- {
-		d := f.field.Directives[i]
+	ds := f.field.PackedDirectives
+	for i := len(ds) - 1; i >= 0; i-- {
+		d := ds[i]
```

Comment:
Can we just do? Probably we need to document order of execution of these directives and it'd be fine... The schema authors would have control by changing the directive order in the schema definition.
```suggestion
for _, d := range f.field.PackedDirectives {
}
```

Reply:
I just realized that [maps in Go do not preserve order](https://go.dev/blog/maps#iteration-order). Should we store these in a slice in the schema? Or at least sort the keys in order?